hyper-lib/src/node.rs

```diff
@@ -853,13 +887,13 @@ impl Node {
                     assert!(peer.get_tx_reconciliation_state().unwrap().is_reconciling(), "Received a reconciliation difference from a peer that we haven't requested to reconcile with (peer_id: {peer_id})");
                     if wants_tx {
                         assert!(self.knows_transaction(), "Received a reconciliation difference from peer (peer_id: {peer_id}) containing a transaction we don't know about");
-                        assert!(!peer.knows_transaction(), "Received a reconciliation difference from peer (peer_id: {peer_id}) containing a transaction they already know");
+                        assert!(!peer.already_announced(), "Received a reconciliation difference from peer (peer_id: {peer_id}) containing a transaction they already know");
                     }
 
                     // If they don't want the transaction, and we have the it in their set, they already know it
                     if !wants_tx && recon_state.get_recon_set() {
                         let peer_mut = self.get_peer_mut(&peer_id).unwrap();
-                        peer_mut.add_known_transaction()
+                        peer_mut.add_tx_announcement(TxAnnouncement::Sent)
```

Comment:
More like "reconciled" but yeah

Reply:
We could have a special case for reconciliation here, but I think it'd be overkill for the current purpose. Will add it if we care about the distinction at some point